manifests/params.pp

```diff
@@ -103,6 +106,7 @@
     'windows' => 'Administrator',
     default => 'root'
   },
+  Optional[String[1]] $default_host          = $facts['clientcert'],
```

Comment:
since the default isn't undef, you cannot pass undef to it. This means we can default to String as datatype:
```suggestion
  String[1] $default_host                    = $facts['clientcert'],
```